### PR TITLE
Authorization credentials

### PR DESCRIPTION
**Pull request by dsimp**

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -67,4 +69,10 @@ def set_comment
     def comment_params
       params.require(:comment).permit(:author_id, :photo_id, :body)
     end
+
+    def is_an_authorized_user
+      @photo = Photo.find(params.fetch(:comment).fetch(:photo_id))
+      if current_user != @photo.owner && @photo.owner.private? && !current_user.leaders.include?(@photo.owner)        redirect_back fallback_location: root_url, alert: "Not authorized"
+      end
+    end
```
Comment (Testell) [inline app/controllers/comments_controller.rb:77]: Nice job securing the comments here!

File: app/controllers/photos_controller.rb [hunk 2]
```diff
@@ -67,4 +72,10 @@ def set_photo
     def photo_params
       params.require(:photo).permit(:image, :comments_count, :likes_count, :caption, :owner_id)
     end
+    
+    def ensure_current_user_is_owner
+      if current_user != @photo.owner
+        redirect_back fallback_location: root_url, alert: "You're not authorized for that."
+      end
+    end
```
Comment (Testell) [inline app/controllers/photos_controller.rb:80]: Nice job using  conventional naming

**Review by Testell — CHANGES_REQUESTED** [review]
Everything looks good so far, you just need to apply the same logic to likes and follow_requests!

**Review by bibimbop123 — APPROVED** [review]
picasso of code

File: app/controllers/comments_controller.rb
Comment (viviancreates) [inline app/controllers/comments_controller.rb:1]: Good use of the before action

File: app/controllers/photos_controller.rb
Comment (viviancreates) [inline app/controllers/photos_controller.rb:1]: Nice, the before action restricts only necessary action

File: app/models/comment.rb
Comment (viviancreates) [inline app/models/comment.rb:1]: Good association

**Review by viviancreates — APPROVED** [review]

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -67,4 +69,10 @@ def set_comment
     def comment_params
       params.require(:comment).permit(:author_id, :photo_id, :body)
     end
+
+    def is_an_authorized_user
+      @photo = Photo.find(params.fetch(:comment).fetch(:photo_id))
+      if current_user != @photo.owner && @photo.owner.private? && !current_user.leaders.include?(@photo.owner)        redirect_back fallback_location: root_url, alert: "Not authorized"
```
Comment (marywebby) [inline app/controllers/comments_controller.rb:75]: make sure the `redirect_back` is on the next line

File: app/controllers/follow_requests_controller.rb [hunk 4]
```diff
@@ -19,7 +19,6 @@ def new
   def edit
```
Comment (marywebby) [inline app/controllers/follow_requests_controller.rb:19]: make sure you are setting the `:is_authorized_user` here at the top of the controller!

File: app/controllers/likes_controller.rb [hunk 5]
```diff
@@ -1,5 +1,12 @@
 class LikesController < ApplicationController
   before_action :set_like, only: %i[ show edit update destroy ]
+  before_action :is_an_authorized_user, only: [:destroy, :create]
+
+  def is_an_authorized_user
```
Comment (marywebby) [inline app/controllers/likes_controller.rb:5]: good!

File: app/controllers/photos_controller.rb [hunk 6]
```diff
@@ -50,10 +50,15 @@ def update
 
   # DELETE /photos/1 or /photos/1.json
   def destroy
-    @photo.destroy
-    respond_to do |format|
-      format.html { redirect_back fallback_location: root_url, notice: "Photo was successfully destroyed." }
-      format.json { head :no_content }
+    if current_user == @photo.owner
```
Comment (marywebby) [inline app/controllers/photos_controller.rb:53]: good! make sure to clean up the indentation and spacing

File: app/policies/photo_policy.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+class PhotoPolicy
```
Comment (marywebby) [inline app/policies/photo_policy.rb:1]: this is for pundit! dont need it here

**Review by marywebby — CHANGES_REQUESTED** [review]
Good work! Please fix the follow request controllers, and make sure to edit your routes too!! Please fix and re-request me!

**Pull request merged**